src/sage/categories/integral_domains.py

```diff
@@ -143,6 +143,38 @@ def is_integral_domain(self, proof=True):
             """
             return True
 
+        def is_field(self, proof=True):
+            r"""
+            Return ``True`` if this ring is a field.
+
+            EXAMPLES::
+
+                sage: ZZ['x'].is_field()
+                False
+            """
+            if self.is_finite():
+                return True
+            if proof:
+                raise NotImplementedError("unable to determine whether or not is a field.")
+            else:
+                return False
```

Comment:
```suggestion
            if proof:
                raise NotImplementedError(f"unable to determine whether or not {self} is a field.")
            return False
```
I think a doctest would be good here.